src/CoreWCF.Primitives/tests/EndpointTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using CoreWCF.Configuration;
+using Microsoft.AspNetCore;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace CoreWCF.Primitives.Tests
+{
+    public class EndpointTests
+    {
+        [ServiceContract]
+        interface IContract1
+        {
+            [OperationContract]
+            void Operation1();
+        }
+
+        [ServiceContract]
+        interface IContract2
+        {
+            [OperationContract]
+            void Operation2();
+        }
+
+        class TwoContractService : IContract1, IContract2
+        {
+            public void Operation1()
+            { }
+
+            public void Operation2()
+            { }
+        }
+
+        class Startup
+        {
+            public void ConfigureServices(IServiceCollection services)
+            {
+                services.AddServiceModelServices();
+            }
+
+            public void Configure(IApplicationBuilder app)
+            {
+                app.UseServiceModel(builder =>
+                {
+                    builder.AddService<TwoContractService>();
+                    var binding = new NetTcpBinding();
+                    var address = "net-tcp://localhost/Test";
+                    builder.AddServiceEndpoint<TwoContractService, IContract1>(binding, address);
+                    builder.AddServiceEndpoint<TwoContractService, IContract2>(binding, address);
+                });
+            }
+        }
+
+        [Fact]
+        public void MultipleEndpointsWithSameListenAddressShouldWork()
+        {
+            var builder = WebHost.CreateDefaultBuilder<Startup>(null);
+            builder.UseNetTcp(8080);
+            var host = builder.Build();
```

Comment:
You don't need to do this. If you look at other tests, we pass 0 as the port to let Kestrel pick one.

Reply:
Ok. Initially I called the method without arguments, but this picks port 808 or something, which is privileged on Linux, making all the Linux tests fail. I'll take a look at the code, but if 0 picks 808, I don't think it will work.